### PR TITLE
Settings polish

### PR DESCRIPTION
**Pull request by lorem--ipsum**
See tasklist here : https://docs.google.com/spreadsheets/d/1twbtmyWKllb2JkeEskiOBz_mTXpYxzhlddVQbucjn1w/edit?pli=1#gid=0

Basically fixed a couple of bugs in the settings, added some features and a bunch of fields to various UIs, and improved the overall look.

### CODE REVIEW

File: src/client/components/immutable-dropdown/immutable-dropdown.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export interface ImmutableDropdownProps<T> extends React.Props<any> {
   equal: (a: T, b: T) => boolean;
   renderItem: (a: T) => string;
   keyItem: (a: T) => any;
-  onChange: (newItem: T, isValid: boolean, path: string) => void;
+  onChange: (newItem: any, isValid: boolean, path: string) => void;
```
Comment (vogievetsky) [inline src/client/components/immutable-dropdown/immutable-dropdown.tsx]: why did you need this change?

Comment (lorem--ipsum) [inline src/client/components/immutable-dropdown/immutable-dropdown.tsx]: I don't need it, but it doesn't make sense before the change. The newItem is not a new list item, it's an "instance". So nothing to do with `<T>`.

File: src/client/components/immutable-input/immutable-input.mocha.tsx [hunk 2]
```diff
@@ -89,6 +89,66 @@ describe('ImmutableInput', () => {
...
+    expect(args[0].clusterName).to.equal('pouet');
+
+    expect(args[1]).to.equal(true);
+
+    expect(args[2]).to.equal('clusterName');
+
+    expect(node.value).to.equal('pouet');
+  });
+
+  describe('with codec (toValue/fromValue)', () => {
```
Comment (vogievetsky) [inline src/client/components/immutable-input/immutable-input.mocha.tsx]: update description here

File: src/client/components/immutable-input/immutable-input.mocha.tsx [hunk 3]
```diff
@@ -89,6 +89,66 @@ describe('ImmutableInput', () => {
...
+    beforeEach(() => {
+      let toValue = (str: string) => str.toLowerCase();
+      let fromValue = (str: string) => str.toUpperCase();
```
Comment (vogievetsky) [inline src/client/components/immutable-input/immutable-input.mocha.tsx]: rename

File: src/client/components/immutable-input/immutable-input.tsx [hunk 4]
```diff
@@ -49,14 +73,37 @@ export class ImmutableInput extends React.Component<ImmutableInputProps, Immutab
...
     });
   }
 
   componentWillReceiveProps(nextProps: ImmutableInputProps) {
-    if (nextProps.instance !== this.state.newInstance) {
+    if (nextProps.instance === undefined) {
+      this.reset();
+    }
+
+    if (this.state.invalidString === undefined && nextProps.instance !== this.state.myInstance) {
```
Comment (vogievetsky) [inline src/client/components/immutable-input/immutable-input.tsx]: maybe use `.equals` here instead of `!==`

File: src/client/views/settings-view/measure-modal/measure-modal.tsx [hunk 5]
```diff
@@ -25,36 +25,43 @@ import { SvgIcon } from '../../../components/svg-icon/svg-icon';
...
 import { Measure } from '../../../../common/models/index';
 
+import { MEASURE_EDIT as LABELS } from '../utils/labels';
+
 
 export interface MeasureModalProps extends React.Props<any> {
   measure?: Measure;
   onSave?: (measure: Measure) => void;
   onClose?: () => void;
+  isCreating?: boolean;
```
Comment (vogievetsky) [inline src/client/views/settings-view/measure-modal/measure-modal.tsx]: rather than a `boolean` this might work better as `mode: "creating" | "editing"`

File: src/client/components/immutable-dropdown/immutable-dropdown.tsx (hunk 1 above)
Comment (vogievetsky) [inline src/client/components/immutable-dropdown/immutable-dropdown.tsx]: oh right. 👍

File: src/client/components/immutable-input/immutable-input.tsx (hunk 4 above)
Comment (lorem--ipsum) [inline src/client/components/immutable-input/immutable-input.tsx]: That wouldn't work. The idea here is to check for strict equality, in case the call to `onChange` failed.

File: src/client/components/simple-table/simple-table.tsx [hunk 6]
```diff
@@ -115,12 +117,13 @@ export class SimpleTable extends React.Component<SimpleTableProps, SimpleTableSt
...
 
   renderRow(row: any, columns: SimpleTableColumn[], index: number): JSX.Element {
+    const { hoveredRowIndex } = this.state;
     var items: JSX.Element[] = [];
 
     for (let i = 0; i < columns.length; i++) {
       let col = columns[i];
 
-      let icon = col.cellIcon ? <SvgIcon className="icon" svg={require(`../../icons/${col.cellIcon}.svg`)}/> : null;
+      let icon = col.cellIcon ? <SvgIcon svg={require(`../../icons/${col.cellIcon}.svg`)}/> : null;
```
Comment (vogievetsky) [inline src/client/components/simple-table/simple-table.tsx]: So I know this is not new code but there is a slight improvement that can be made here: webpack is smart(ish) when it sees code like `require(`../../icons/${col.cellIcon}.svg` it thinks to itself ok cool this might be any file that is `../../icons/*.svg` better depend on all of them to be safe.
It would be better to have a prefix for the icons. Something like: `require(`../../icons/table-${col.cellIcon}.svg` (and rename the icon files accordingly) that way only files with this prefix will be dependent on here. see: https://github.com/implydata/pivot/blob/master/src/client/components/vis-selector-menu/vis-selector-menu.tsx#L88

**Comment by vogievetsky** [comment]
gr 🎱
